### PR TITLE
Update specviz parser to handle SpectrumList input

### PR DESCRIPTION
**Pull request by rosteen**
### Description

Intended to address #822. Depends on https://github.com/astropy/specutils/pull/856 for loading the most recent JWST multi-extension data files. I tested with the following files from @havok2063's folder:

- jw00721-o023_t008_nircam_f444w-grismr-subgrism256_x1dints.fits
- jw00787-o014_s00033_niriss_f150w-gr150c-gr150r_x1d.fits
- jw00787-o014_s00015_niriss_f150w-gr150c-gr150r_c1d.fits

I'm also planning to add batch linking a la Mosviz now that #762 is merged.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by ibusko — APPROVED** [review]
Tested with a bunch of files from @havok test data repo. LGTM.

File: jdaviz/configs/specviz/plugins/parsers.py [hunk 1]
```diff
@@ -30,32 +32,52 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
...
-
-    app.add_data(data, data_label)
-    if show_in_viewer:
-        app.add_data_to_viewer("spectrum-viewer", data_label)
+    with app.data_collection.delay_link_manager_update():
+        for i in range(len(data)):
+            spec = data[i]
+            if current_spec != {} and current_spec is not None:
+                spec_key = list(current_spec.keys())[0]
+                current_unit = current_spec[spec_key].spectral_axis.unit
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/parsers.py:73]: I don't understand why `current_spec` logic is in a loop. As far as I can see, it doesn't change per iteration. Can we move this out of the loop? Did I miss something?

File: jdaviz/configs/specviz/plugins/parsers.py [hunk 2]
```diff
@@ -30,32 +32,52 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
...
+                if spec.spectral_axis.unit != current_unit:
+                    spec = Spectrum1D(flux=spec.flux,
+                                      spectral_axis=spec.spectral_axis.to(current_unit))
+
+            app.add_data(spec, data_label[i])
+            if show_in_viewer:
+                app.add_data_to_viewer("spectrum-viewer", data_label[i])
+
+            # Only auto-show the first spectrum in a list
+            show_in_viewer = False
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/parsers.py:83]: Same result but less lines?

```suggestion
            # Only auto-show the first spectrum in a list
            if i == 0 and show_in_viewer:
                app.add_data_to_viewer("spectrum-viewer", data_label[i])
```

File: jdaviz/configs/specviz/tests/test_helper.py [hunk 3]
```diff
@@ -12,6 +12,7 @@ class TestSpecvizHelper:
     def setup_class(self, specviz_app, spectrum1d):
         self.spec_app = specviz_app
         self.spec = spectrum1d
+        self.spec_list = SpectrumList([spectrum1d]*3)
```
Comment (pllim) [inline jdaviz/configs/specviz/tests/test_helper.py:15]: Total nitpick.
```suggestion
        self.spec_list = SpectrumList([spectrum1d] * 3)
```

File: jdaviz/configs/specviz/plugins/parsers.py [hunk 4]
```diff
@@ -30,32 +32,52 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
             base64.b85encode(uuid.uuid4().bytes), "utf-8"
         )
 
-    # If data provided is a path, try opening into a Spectrum1D object
-    try:
+    if isinstance(data, SpectrumCollection):
+        raise TypeError("SpectrumCollection detected."
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/parsers.py:36]: Might as well test this too?

**Review by pllim — COMMENTED** [review]
Do you also plan to include `SpectrumList` usage in specviz example notebook?

File: jdaviz/configs/specviz/plugins/parsers.py (hunk 2 above)
Comment (rosteen) [reply jdaviz/configs/specviz/plugins/parsers.py]: Whoops, apparently I can't commit this from here now.

**Comment by rosteen** [comment]
@pllim thanks for the suggestions, I addressed everything you called out.

**Review by pllim — APPROVED** [review]
LGTM thanks!

**Comment by pllim** [comment]
Since this depends on astropy/specutils#856 , do we need another specutils release, or does that comment on the original post no longer apply?

**Comment by rosteen** [comment]
I think we can merge, as the feature is generally applicable. It just won't work to load some particular JWST files without a specutils release (which I can do tomorrow).

**Comment by havok2063** [comment]
@rosteen which viztool should be used to open these files?

**Comment by pllim** [comment]
Oh no... I think this broke mosviz...

```python
mosviz.load_data(directory=data_dir, instrument="nirspec")
```

now gives

`The number of images in this directory does not match the number of spectra 1d and 2d files, please make the amounts equal or load images separately.`

**Comment by pllim** [comment]
(I opened #851 so we don't forget about it.)

**Comment by javerbukh** [comment]
@pllim This may not work for this issue but putting it out there for posterity, can you run `rm -R mosviz_nirspec_data_0.3/` in your temp directory? That will remove a previous version of that directory that contains a different number of images from spectra.

**Comment by rosteen** [comment]
@havok2063 This is for Specviz. 

@pllim Pretty sure that's unrelated to this PR, it's because the zip file loaded there doesn't have the same number of images as it does spectra. I'll comment over in #851 as well.

**Pull request merged**